### DISCUSSION TITLE
How to make urql throw an error when there is an error?

### DISCUSSION THREAD

**Discussion by thehappycoder** [urql-graphql/urql#1436] [Q&A] [answered]
I know graphql response could be partially erroneous but still need it to throw.

Here is what I am doing for now:
```
const response = await client
    .mutation(mutation, {
      input: {
        message,
        transactionIds,
      },
    })
    .toPromise();

  if (response.error) {
    throw response.error;
  }

  return response.data;
```

**Comment by kitten** [comment, marked as answer]
If you need to reject the promise then that's exactly what we recommend; choose the error condition yourself and throw since that replaces any other option or configuration which would make it less intuitive
